src/nvim/plines.c

```diff
@@ -991,25 +994,43 @@ int64_t win_text_height(win_T *const wp, const linenr_T start_lnum, const int64_
     lnum = lnum_next + 1;
   }
 
-  while (lnum <= end_lnum) {
+  while (lnum <= *end_lnum && height_sum_nofill + height_sum_fill < max) {
     linenr_T lnum_next = lnum;
-    hasFolding(wp, lnum, &lnum, &lnum_next);
-    height_sum_fill += win_get_fill(wp, lnum);
+    folded_end = hasFolding(wp, lnum, &lnum, &lnum_next);
+    fill_end = win_get_fill(wp, lnum);
+    height_sum_fill += fill_end;
     height_cur_nofill = plines_win_nofill(wp, lnum, false);
     height_sum_nofill += height_cur_nofill;
+    lnum_end = lnum;
     lnum = lnum_next + 1;
   }
 
-  if (end_vcol >= 0) {
+  int64_t vcol_end = *end_vcol;
+  bool use_vcol = vcol_end >= 0 && lnum_end == *end_lnum;
+  if (use_vcol) {
```

Comment:
This isn't correct. `lnum_end == *end_lnum` is `false` when the last `*end_lnum` is at the end of the fold, but that make it impossible to only count filler lines above the fold without counting the fold itself. I think it should be changed to `lnum > *end_lnum` instead.

Reply:
Why? If `lnum_end != *end_lnum`, `vcol_end` has no meaning. I don't see how the fact that statement isn't true because `*end_lnum` was part of a closed fold changes that. It's documented to be the virtual column index on `"end_row"`.

TBF I'm not quite sure what you mean with "only count filler lines above the fold without counting the fold itself" becoming impossible, or why it is important to maintain that possibility. If it is, I suppose `end_vcol` should document that it applies to the end of a closed fold if `end_row` points to a closed fold region.

Reply:
I'm trying to add test for this in #33529. The last line of the test is what I'm referring to.

Reply:
Yeah I get it now, makes sense I think. `end_vcol` having no meaning on the folded line is caught by `MIN(row_off, height_cur_nofill)` anyways.